storage/range.go

```diff
@@ -169,7 +175,26 @@ func NewRange(desc *proto.RangeDescriptor, rm RangeManager) *Range {
 		respCache:   NewResponseCache(desc.RaftID, rm.Engine()),
 		pendingCmds: map[cmdIDKey]*pendingCmd{},
 	}
-	return r
+
+	// Look in the engine to find the last log index.
+	logKey := engine.RaftLogPrefix(uint64(r.Desc.RaftID))
```

Comment:
Moved to Range.loadLastIndex. And per our earlier conversation about the expense of doing this at startup, note that this is only looking for the key of the first (i.e. most recent) log entry; we don't need to load the entire log at startup. If this turns out to be too expensive because it does a lot of seeks, we could perhaps write lastIndex to an area next to the HardState when we append new entries.